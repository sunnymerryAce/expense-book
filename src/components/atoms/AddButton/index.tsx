import React from "react";
import Fab from "@material-ui/core/Fab";
import AddIcon from "@material-ui/icons/Add";
interface Props {
  onClick: (props: any) => any;
}

const AddButton: React.FC<Props> = (props) => {
  return (
    <Fab onClick={props.onClick} color="secondary">
      <AddIcon />
    </Fab>
  );
};

export default AddButton;
