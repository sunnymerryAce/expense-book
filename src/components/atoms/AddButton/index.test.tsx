import React from "react";
import AddButton from "./";
import { shallow } from "enzyme";

describe("<AddButton />", () => {
  let container = null;
  const onClickFunction = jest.fn();

  beforeEach(() => {
    container = shallow(<AddButton handleClick={onClickFunction} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    expect(container.find(".AddButton").props()).toMatchObject({
      onClick: onClickFunction,
    });
  });

  test("click event handler should be triggered", () => {
    container.simulate("click");
    expect(onClickFunction).toBeCalled();
  });
});
