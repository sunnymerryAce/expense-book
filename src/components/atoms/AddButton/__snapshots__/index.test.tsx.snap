// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddButton /> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiFab-root AddButton MuiFab-secondary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiFab-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></span></button>"`;
